ui/perfherder/helpers.js

```diff
@@ -377,3 +389,407 @@ export const getGraphsLink = function getGraphsLink(
 
   return `perf.html#/graphs${createQueryParams(params)}`;
 };
+
+// old PhAlerts' inner workings
+const Alert = (alertData, optionCollectionMap) => ({
+  ...alertData,
+  title: getSeriesName(alertData.series_signature, optionCollectionMap),
+});
+
+export const getAlertStatusText = alert =>
+  Object.values(phAlertStatusMap).find(status => status.id === alert.status)
+    .text;
+
+export const getGraphsURL = (
+  alert,
+  timeRange,
+  alertRepository,
+  performanceFrameworkId,
+) => {
+  let url = `#/graphs?timerange=${timeRange}&series=${alertRepository},${
+    alert.series_signature.id
+  },1`;
+
+  // automatically add related branches (we take advantage of
+  // the otherwise rather useless signature hash to avoid having to fetch this
+  // information from the server)
+  if (phFrameworksWithRelatedBranches.includes(performanceFrameworkId)) {
+    const branches =
+      alertRepository === 'mozilla-beta'
+        ? ['mozilla-inbound']
+        : thPerformanceBranches.filter(branch => branch !== alertRepository);
+    url += branches
+      .map(
+        branch =>
+          `&series=${branch},${alert.series_signature.signature_hash},0`,
+      )
+      .join('');
+  }
+
+  return url;
+};
+
+export const getSubtestsURL = (alert, alertSummary) => {
+  const endpoint = '#/comparesubtest';
+  const urlParameters = {
+    framework: alertSummary.framework,
+    originalProject: alertSummary.repository,
+    originalSignature: alert.series_signature.signature_hash,
+    newProject: alertSummary.repository,
+    newSignature: alert.series_signature.signature_hash,
+  };
+  if (alertSummary.prevResultSetMetadata) {
+    urlParameters.originalRevision =
+      alertSummary.prevResultSetMetadata.revision;
+  }
+  if (alertSummary.prevResultSetMetadata) {
+    urlParameters.newRevision = alertSummary.resultSetMetadata.revision;
+  }
+
+  return `${endpoint}${createQueryParams(urlParameters)}`;
+};
+
+const modifyAlert = (alert, modification) =>
+  update(getApiUrl(`/performance/alert/${alert.id}/`), modification);
+
+export const alertIsOfState = (alert, phAlertStatus) =>
+  alert.status === phAlertStatus.id;
+
+export const toggleStar = alert => {
+  const toggledStar = !alert.starred;
+  modifyAlert(alert, {
+    starred: toggledStar,
+  }).then(() => {
+    alert.starred = toggledStar;
+  });
+};
+
+let issueTrackers; // will cache on first AlertSummary call
+
+const _initializeAlerts = (alertSummary, optionCollectionMap) => {
+  // this function converts the representation returned by the perfherder
+  // api into a representation more suited for display in the UI
+
+  // just treat related (reassigned or downstream) alerts as one
+  // big block -- we'll display in the UI depending on their content
+  alertSummary.alerts = alertSummary.alerts
```

Comment:
I must admit, I'm not usually a fan of a function that modifies one of the passed-in parameters.  Would it be possible to make this a ``get``er and modify ``alertSummary.alerts`` with the result of this function?